--- volumes.py	2015-08-24 15:55:33.800287495 +0800
+++ volumes.new.py	2015-08-24 16:03:39.816531404 +0800
@@ -16,6 +16,8 @@
 """Volume interface (v2 extension)."""
 
 import six
+from heat.openstack.common import log as logging
+LOG = logging.getLogger(__name__)
 try:
     from urllib import urlencode
 except ImportError:
@@ -38,6 +40,9 @@
         """Delete this volume."""
         self.manager.delete(self)
 
+    def soft_delete(self):
+       self.manager.soft_delete(self)
+
     def update(self, **kwargs):
         """Update the name or description for this volume."""
         self.manager.update(self, **kwargs)
@@ -170,7 +175,7 @@
                volume_type=None, user_id=None,
                project_id=None, availability_zone=None,
                metadata=None, imageRef=None, scheduler_hints=None,
-               source_replica=None):
+               source_replica=None, host_name=None):
         """Creates a volume.
 
         :param size: Size of volume in GB
@@ -211,11 +216,10 @@
                            'imageRef': imageRef,
                            'source_volid': source_volid,
                            'source_replica': source_replica,
+                           'host_name': host_name,
                            }}
-
         if scheduler_hints:
             body['OS-SCH-HNT:scheduler_hints'] = scheduler_hints
-
         return self._create('/volumes', body, 'volume')
 
     def get(self, volume_id):
@@ -290,6 +294,10 @@
         """
         self._delete("/volumes/%s" % base.getid(volume))
 
+    
+    def soft_delete(self, volume):
+        self._soft_delete("/volumes/%s/action" % base.getid(volume))
+ 
     def update(self, volume, **kwargs):
         """Update the name or description for a volume.
 
